### PR TITLE
Modify query SQL for MySQLDAO

### PR DESCRIPTION
**Pull request by tristaZero**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [x] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by wu-sheng** [comment]
Are these SQL still different with the H2?

File: oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/mysql/MySQLAggregationQueryDAO.java [hunk 1]
```diff
@@ -44,14 +49,14 @@ public MySQLAggregationQueryDAO(
...
-        sql.append("select * from (select avg(").append(valueCName).append(") value,").append(Metrics.ENTITY_ID).append(" from ")
+        sql.append("select avg(").append(valueCName).append(") value,").append(Metrics.ENTITY_ID).append(" from ")
             .append(tableName).append(" where ");
         this.setTimeRangeCondition(sql, conditions, startTB, endTB);
         if (appender != null) {
             appender.append(sql, conditions);
         }
         sql.append(" group by ").append(Metrics.ENTITY_ID);
-        sql.append(") AS METRICS order by value ").append(order.equals(Order.ASC) ? "asc" : "desc").append(" limit ").append(topN);
+        sql.append(" order by value ").append(order.equals(Order.ASC) ? "asc" : "desc").append(" limit ").append(topN);
```
Comment (wu-sheng) [inline oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/mysql/MySQLAggregationQueryDAO.java]: Have you used real data set to verify this?

Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/mysql/MySQLAggregationQueryDAO.java]: Could you post the test data set and result in this PR?

Comment (tristaZero) [reply oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/mysql/MySQLAggregationQueryDAO.java]: Add Unit test to thi pr? Or just comment my test result?

Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/mysql/MySQLAggregationQueryDAO.java]: Comment should be enough.

**Comment by tristaZero** [comment]
Here are test results,
```
##### The original SQL:
select * from (select avg(value) value,entity_id from endpoint_avg where time_bucket >= ? and time_bucket <= ? group by entity_id) AS METRICS order by value desc limit 10
##### The rewritten SQL:
select avg(value) value,entity_id from endpoint_avg where time_bucket >= ? and time_bucket <= ? group by entity_id order by value desc limit 10

##### SQLss similar with the above ones,

mysql> select * from (select avg(item_id) value,order_id from t_order_item_1 group by order_id) tb order by value desc limit 2;
+--------+----------+
| value  | order_id |
+--------+----------+
| 8.0000 |        2 |
| 6.3333 |        1 |
+--------+----------+
2 rows in set (0.01 sec)

mysql> select avg(item_id) value, order_id from t_order_item_1 group by order_id order by value desc limit 2;
+--------+----------+
| value  | order_id |
+--------+----------+
| 8.0000 |        2 |
| 6.3333 |        1 |
+--------+----------+
2 rows in set (0.00 sec)

**Comment by wu-sheng** [comment]
@tristaZero Still one question, are these SQLs same with H2 or not? Could them be merged together?

**Comment by tristaZero** [comment]
@wu-sheng Hi, i think so. Actually there are 4 `MySQLXXXXDAO` overwriting some functions of `H2XXXDAO`,  i.e., 
```
MySQLAggregationQueryDAO
MySQLTraceQueryDAO
MySQLAlarmQueryDAO
MySQLLogQueryDAO
```
And i compared with those overwrite functions and found it seems that they return the SAME SQLs. I do not know why those `MySQLXXXDAO` were created, howerver maybe we can try to use `H2XXXDAO` for `MySQLProvider` and `H2Provider` uniformly.

**Comment by wu-sheng** [comment]
It they are using the same SQL, they should be merged. I remember they are different, but not sure. If you are sure, please process that.

**Comment by tristaZero** [comment]
@wu-sheng Hi Try best to make `MySQLDao` become simplier and correct.

**Comment by kezhenxu94** [comment]
> @kezhenxu94 Do we have e2e for H2 storage?

no now

**Comment by tristaZero** [comment]
@kezhenxu94, Thanks feedback, i will review those log after my airplane arrives in shenzhen...

File: oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/h2/dao/H2AggregationQueryDAO.java [hunk 2]
```diff
@@ -83,18 +86,17 @@ public H2AggregationQueryDAO(JDBCHikariCPClient h2Client) {
 
     public List<TopNEntity> topNQuery(String indName, String valueCName, int topN, Downsampling downsampling,
                                       long startTB, long endTB, Order order, AppendCondition appender) throws IOException {
-        String tableName = ModelName.build(downsampling, indName);
+        String indexName = ModelName.build(downsampling, indName);
         StringBuilder sql = new StringBuilder();
         List<Object> conditions = new ArrayList<>(10);
-        sql.append("select avg(").append(valueCName).append(") value,").append(Metrics.ENTITY_ID).append(" from ")
-                .append(tableName).append(" where ");
+        sql.append("select * from (select avg(").append(valueCName).append(") value,").append(Metrics.ENTITY_ID).append(" from ")
```
Comment (wu-sheng) [inline oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/h2/dao/H2AggregationQueryDAO.java]: This kind of query statement is required in H2?

**Comment by tristaZero** [comment]
@wu-sheng, Hi all 4 MySQLXXDAO are same with H2XXDAO. however, i found those SQL is redundant for MySQL and could be simple, on the other hand, as the errorlog shows, `any item in order by list` must be in group by list. Therefore, this kind of SQL is required For H2, but not for MySQL.

**Comment by wu-sheng** [comment]
Look like you just break all checks :) Please recheck locally.

**Comment by tristaZero** [comment]
@wu-sheng Hi, i read the error log, however the error context confused me and i am not sure those my code changes caused those exception:
```
Using context: Agent Plugin Tests Workload 3

ERROR: /x1/jenkins/jenkins-home/jobs/skywalking-agent-test-workload-3/workspace@script/Jenkinsfile-Agent-Test-3 not found

Finished: FAILURE
Using context: e2e

ERROR: /x1/jenkins/jenkins-home/jobs/skywalking-e2e-2/workspace@script/Jenkinsfile-E2E not found

Finished: FAILURE
```
Could you give me some advices? Thanks.

**Comment by tristaZero** [comment]
It's a problem of code styple? I tried to load codeStyple.xml in root path, but IDEA told me something like ,
```
com.puppycrawl.tools.checkstyle.api.CheckstyleException: unable to parse configuration stream - Document is invalid: no grammar found.:19:13
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:343)
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:308)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.infernus.idea.checkstyle.service.cmd.OpLoadConfiguration.loadConfigurationForCheckstylePre825(OpLoadConfiguration.java:173)
	at org.infernus.idea.checkstyle.service.cmd.OpLoadConfiguration.callLoadConfiguration(OpLoadConfiguration.java:156)
	at org.infernus.idea.checkstyle.service.cmd.OpLoadConfiguration.execute(OpLoadConfiguration.java:139)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.loadConfig(OpCreateChecker.java:91)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.execute(OpCreateChecker.java:54)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.execute(OpCreateChecker.java:26)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:130)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.createChecker(CheckstyleActionsImpl.java:60)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.createChecker(CheckstyleActionsImpl.java:51)
	at org.infernus.idea.checkstyle.checker.CheckerFactoryWorker.run(CheckerFactoryWorker.java:46)
Caused by: org.xml.sax.SAXParseException; lineNumber: 19; columnNumber: 13; Document is invalid: no grammar found.
	at org.apache.xerces.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source)
	at org.apache.xerces.util.ErrorHandlerWrapper.error(Unknown Source)
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
	at org.apache.xerces.impl.XMLNSDocumentScannerImpl.scanStartElement(Unknown Source)
	at org.apache.xerces.impl.XMLNSDocumentScannerImpl$NSContentDispatcher.scanRootElementHook(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at com.puppycrawl.tools.checkstyle.XmlLoader.parseInputSource(XmlLoader.java:99)
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.parseInputSource(ConfigurationLoader.java:189)
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:336)
	... 15 more
```

**Comment by kezhenxu94** [comment]
> @wu-sheng Hi, i read the error log, however the error context confused me and i am not sure whether my code changes caused those exception:
> 
> ```
> Using context: Agent Plugin Tests Workload 3

It's issue of code styles, but don't refer to the Jenkins jobs, they are abandoned, refer to the Github Actions' log, the reported error seems to be:

```text
2019-12-17T03:05:34.8641424Z [ERROR] /home/runner/work/skywalking/skywalking/oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/mysql/MySQLLogQueryDAO.java:37: 'method def rcurly' has incorrect indentation level 0, expected level should be 4. [Indentation]
2019-12-17T03:05:34.8641820Z Audit done.
```

>  I tried to load codeStyple.xml in root path, but IDEA told me something like ,

Did you import the `codeStyle.xml` as configuration of `checkstyle` plugin in IDEA, `codeStyle.xml` is for IDEA itself AFAIK, not for `checkstyle` plugin

**Comment by wu-sheng** [comment]
@tristaZero Same as @kezhenxu94 's comments. Ignore all jenkins jobs. SkyWalking has moved in GitHub Action now.

**Comment by tristaZero** [comment]
@kezhenxu94 Hi zhenxu, which checkstyple.xml is correct one for `checkstyple plugin`? I tried to add `/apm-checkstyle/checkStyple.xml` for `checkstyple plugin setting`, however i got exception as follows:
```
com.puppycrawl.tools.checkstyle.api.CheckstyleException: unable to parse configuration stream
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:346)
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:308)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.infernus.idea.checkstyle.service.cmd.OpLoadConfiguration.loadConfigurationForCheckstylePre825(OpLoadConfiguration.java:173)
	at org.infernus.idea.checkstyle.service.cmd.OpLoadConfiguration.callLoadConfiguration(OpLoadConfiguration.java:156)
	at org.infernus.idea.checkstyle.service.cmd.OpLoadConfiguration.execute(OpLoadConfiguration.java:139)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.loadConfig(OpCreateChecker.java:91)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.execute(OpCreateChecker.java:54)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.execute(OpCreateChecker.java:26)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:130)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.createChecker(CheckstyleActionsImpl.java:60)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.createChecker(CheckstyleActionsImpl.java:51)
	at org.infernus.idea.checkstyle.checker.CheckerFactoryWorker.run(CheckerFactoryWorker.java:46)
Caused by: org.xml.sax.SAXException: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Property ${checkstyle.header.file} has not been set
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Property ${checkstyle.header.file} has not been set
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader.startElement(ConfigurationLoader.java:649)
	at org.apache.xerces.parsers.AbstractSAXParser.startElement(Unknown Source)
	at org.apache.xerces.parsers.AbstractXMLDocumentParser.emptyElement(Unknown Source)
	at org.apache.xerces.impl.dtd.XMLDTDValidator.emptyElement(Unknown Source)
	at org.apache.xerces.impl.XMLNSDocumentScannerImpl.scanStartElement(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
	at com.puppycrawl.tools.checkstyle.XmlLoader.parseInputSource(XmlLoader.java:99)
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.parseInputSource(ConfigurationLoader.java:189)
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:336)
	... 15 more
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Property ${checkstyle.header.file} has not been set
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.replaceProperties(ConfigurationLoader.java:495)
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.access$500(ConfigurationLoader.java:53)
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader.startElement(ConfigurationLoader.java:644)
	... 29 more

```

**Comment by kezhenxu94** [comment]
> @kezhenxu94 Hi zhenxu, which checkstyple.xml is correct one for `checkstyple plugin`? I tried to add `/apm-checkstyle/checkStyple.xml` for `checkstyple plugin setting`, however i got exception as follows:

`/apm-checkstyle/checkStyple.xml` is the right one, but the reported exception does no harm, just ignore it because of the plugin cannot recognize `${checkstyle.header.file}`, it's for license header checks

For this case, I think reformatting the file `MySQLLogQueryDAO.java` should work

**Comment by tristaZero** [comment]
> > @kezhenxu94 Hi zhenxu, which checkstyple.xml is correct one for `checkstyple plugin`? I tried to add `/apm-checkstyle/checkStyple.xml` for `checkstyple plugin setting`, however i got exception as follows:
> 
> `/apm-checkstyle/checkStyple.xml` is the right one, but the reported exception does no harm, just ignore it because the plugin cannot recognize `${checkstyle.header.file}`, it's for license header checks
> 
> For this case, I think reformatting the file `MySQLLogQueryDAO.java` should work

@kezhenxu94 I bet you are my sweetheart!

**Pull request merged**